server/controllers/documentController.js

```diff
@@ -1,132 +1,298 @@
+          results.forEach((document) => {
+            if (Role === 1) {
+              searchResult.push(document.dataValues);
+            } else if ((document.access === 'public' ||
+              document.user.dataValues.RoleId === request.decoded.RoleId)
+              && document.access !== 'private') {
+              searchResult.push(document.dataValues);
+            } else if (document.access === 'private' && document.OwnerId === request.decoded.UserId) {
+              searchResult.push(document.dataValues);
             }
-        })
-            .then(document => {
-                if (document) {
-                    document.destroy()
-                        .then(() => response.status(201).send({
-                            success: true,
-                            message: 'Document has been successfully deleted'
-                        }))
-                        .catch(error => response.status(401).send(error))
-                } else {
-                    response.status(404).send({
-                        success: false,
-                        message: 'Document not found'
-                    })
-                }
-            })
+          });
+          response.status(302).send({
+            success: true,
+            message: `Documents matching ${searchQuery}`,
+            documents: searchResult
+          });
+        } else {
+          response.status(401).send({
+            success: false,
+            message: 'Document not found'
+          });
+        }
+      })
+      .catch((error) => {
+        response.status(401).send({
+          success: false,
+          message: error.message
+        });
+      });
+  }
+  /**
+   * Fetch all the documents belonging to a particular user
+   * Users have access to their own documents and all other public and role access documents
+   * @param{Object} request - Server request
+   * @param{Object} response - Server response
+   * @return{Void} - return Void
+   */
+  static fetchUserDocument(request, response) {
+    const userRole = request.decoded.RoleId;
+    const Owner = request.decoded.UserId;
+    const queryBuilder = {
+      attributes: ['id', 'email', 'firstName', 'lastName'],
+      include: [{
+        model: Documents,
+        attributes: ['id', 'title', 'content', 'createdAt'],
+      }]
+    };
+    Users.findById(request.params.id, queryBuilder)
+      .then((results) => {
+        if (results) {
+          const result = results.dataValues.documents;
+          const searchResult = [];
+          result.forEach((document) => {
+            if (userRole === 1) {
+              searchResult.push(document.dataValues);
+            } else if ((document.access === 'public' ||
+              document.user.dataValues.RoleId === userRole)
+              && document.access !== 'private') {
+              searchResult.push(document.dataValues);
+            } else if (document.access === 'private' && document.OwnerId === Owner) {
```

Comment:
to avoid confusion, always use the same type of data when comparing. E.g, instead of `document.access === 'private' && document.OwnerId === Owner` use `document.access === 'private' && document.OwnerId === OwnerId`. This will make it clear that you are comparing the user ids as to thinking you are comparing an id to a Owner object.